aws-android-sdk-core/src/test/java/com/amazonaws/mobile/config/AWSConfigurationTest.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *    http://aws.amazon.com/apache2.0
+ *
+ * This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.amazonaws.mobile.config;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+
+public class AWSConfigurationTest {
+
+    @Test
+    public void testAWSConfigurationWithJSONObject() {
```

Comment:
Nitpick: It might be a better test to try to construct a `CognitoCachingCredentialsProvider` to make sure at least one client reads it as expected.

Reply:
Added